blocks/bricks/recurrent.py

```diff
@@ -185,7 +187,7 @@ def scan_function(*args):
                 args = list(args)
                 arg_names = (list(sequences_given) + list(states_given) +
                              list(contexts_given))
-                kwargs = dict(equizip(arg_names, args))
+                kwargs = dict(zip(arg_names, args))
```

Comment:
Oh I got it, you want to get rid to trailing arguments. But this is not the nicest way to do it, replacing a safe zip with a completely unsafe one. You should rather do `equizip(arg_names, args[-len(shared_vars):])`.